lib/perl/Genome/Process.pm

```diff
@@ -630,5 +633,97 @@ sub result_is_on_cle_disk_group {
     return Genome::Disk::Group::Validate::GenomeDiskGroups::is_cle_disk_group_name($disk_group_name);
 }
 
+sub compare_output {
+    my ($self, $other_process_id) = @_;
+    my $process_id = $self->id;
+
+    unless (defined ($other_process_id)) {
+        die $self->error_message('Require process ID argument!');
+    }
+    my $other_process = Genome::Process->get($other_process_id);
+    unless ($other_process) {
+        die $self->error_message('Could not get process for ID (%s)', $other_process_id);
+    }
+
+    unless ($self->class eq $other_process->class) {
+        die $self->error_message('Processes (%s) and (%s) are not of the same type', $process_id, $other_process_id);
+    }
+
+    my %diffs = $self->_compare_output_files($other_process);
+
+    return %diffs;
+}
+
+sub _compare_output_files {
```

Comment:
The child classes don't have a single output directory themselves. They merely keep track of the software results they create. The symlink command takes care to aggregate all of the output in one location. E.g., for Genome::VariantReporting::Process::CreateReport, there is one software result for each report. The symlink_results method in that class symlinks all of the files for those software results into one output_directory of your choosing.